File: frontend/src/services/aiService.ts
import type {
  ApiResponse,
  GenerateResumePayload,
  GenerateResumeResponse,
  ImproveBulletPayload,
  ImproveBulletResponse,
  ImproveResumeResponse,
  JobAnalysisPayload,
  JobAnalysisResponse,
} from "@/types";

import apiClient from "./apiClient";

export const aiService = {
  improveBullet: (payload: ImproveBulletPayload) =>
    apiClient.post<ApiResponse<ImproveBulletResponse>>("/ai/improve-bullet", payload),

  generateSummary: (resumeId: string) =>
    apiClient.post<ApiResponse<{ summary: string }>>("/ai/generate-summary", { resumeId }),

  suggestSkills: (resumeId: string) =>
    apiClient.post<ApiResponse<{ skills: string[] }>>("/ai/suggest-skills", { resumeId }),

  generateResume: (payload: GenerateResumePayload) =>
    apiClient.post<ApiResponse<GenerateResumeResponse>>("/ai/generate-resume", payload),

  improveResume: (resumeId: string) =>
    apiClient.post<ApiResponse<ImproveResumeResponse>>("/ai/improve-resume", { resumeId }),

  analyzeJob: (payload: JobAnalysisPayload) =>
    apiClient.post<ApiResponse<JobAnalysisResponse>>("/ai/analyze-job", payload),
};

File: backend/src/main/java/com/app/ai/service/AiService.java
package com.app.ai.service;

import com.app.ai.dto.GenerateResumeRequest;
import com.app.ai.dto.GenerateResumeResponse;
import com.app.ai.dto.ImproveBulletRequest;
import com.app.ai.dto.ImproveBulletResponse;
import com.app.ai.dto.ImproveResumeRequest;
import com.app.ai.dto.ImproveResumeResponse;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface AiService {

    /** Generate a resume from a free-text background description. Returns sections map. */
    Map<String, Object> generateResumeFromPrompt(UUID userId, String prompt);

    /** Generate a complete resume from structured fields (job title, level, skills). */
    GenerateResumeResponse generateResume(UUID userId, GenerateResumeRequest request);

    /** Improve an existing resume against a target job description. */
    ImproveResumeResponse improveResume(UUID userId, ImproveResumeRequest request);

    /** Generate a professional summary for the given resume. */
    String generateSummary(UUID userId, UUID resumeId);

    /** Suggest additional skills based on the given resume's content. */
    List<String> suggestSkills(UUID userId, UUID resumeId);

    /** Rewrite a single bullet point using the CAR framework. */
    ImproveBulletResponse improveBullet(UUID userId, ImproveBulletRequest request);
}
